### DISCUSSION TITLE
Problem with throw type routes

### DISCUSSION THREAD

**Discussion by robertdahlem** [strongswan/strongswan#825] [Q&A]
Hi,

I'm running Strongswan 5.7.2 under Debian 10 (buster) and 5.9.1 under Debian 11 (bullseye). I use `load = yes` in `/etc/strongswan.d/charon/bypass-lan.conf`

I noticed that I sporadically seem to lose bypass routes in table 220 under **Debian 11**. Can't say what exactly makes this happen but I can trigger the behavior reproducibly by restarting systemd-networkd:
<pre>
# ip route sh table 220
throw 10.10.10.0/24 proto static
1.2.3.4 via 192.168.1.1 dev ens18 proto static src 172.29.254.11
throw 192.168.1.0/24 proto static
throw 192.168.180.0/24 proto static

# systemctl restart systemd-networkd
# ip route sh table 220
1.2.3.4 via 192.168.1.1 dev ens18 proto static src 172.29.254.11
</pre>

**Debian 10**:
<pre>
# ip route sh table 220
10.10.10.0/24 dev vmbr1 proto static src 10.10.10.1
192.168.1.0/24 dev ens18 proto static src 192.168.1.158
192.168.180.0/24 dev ens19 proto static src 192.168.180.1

# systemctl restart systemd-networkd
# ip route sh table 220
10.10.10.0/24 dev vmbr1 proto static src 10.10.10.1
192.168.1.0/24 dev ens18 proto static src 192.168.1.158
192.168.180.0/24 dev ens19 proto static src 192.168.180.1
</pre>

This is probably due to feature [3118](https://wiki.strongswan.org/issues/3118) ("Use throw type routes instead of copying routes into table 220 for passthrough policies").

It looks like `throw` routes get thrown away (pun intended) by Linux when something happens to the interfaces for short times.  Static routes are more hardy. Strongswan seems not to notice when a bypass route is missing or when an interface goes MIA for a short time.

Is there a way to make this more poedictable?

**Comment by tobiasbrunner** [comment]
Your explanation makes no sense as the throw routes specifically don't reference any interfaces! (The whole point of using them is that we don't have to do that and are not affected by interface changes etc.)

Could be an issue with systemd-networkd removing these routes when it shouldn't (I wouldn't expect the kernel to remove them on its own).

**Reply by robertdahlem** [reply]
Thank you! You are right, Systemd-Networkd removes the routes.

I asked the Systemd developers about it, see my [mail](https://lists.freedesktop.org/archives/systemd-devel/2021-December/047205.html) on [systemd-devel.](https://lists.freedesktop.org/mailman/listinfo/systemd-devel) I will report here when I hear from them.

**Reply by robertdahlem** [reply]
[Turns out](https://lists.freedesktop.org/archives/systemd-devel/2021-December/047208.html): Systemd has a concept of "foreign routes" meaning routes not established by Networkd. By default, these get removed.

To prevent Networkd from doing so: edit `/etc/systemd/networkd.conf` or your own .conf file in `networkd.conf.d`
```
[Network]
ManageForeignRoutes=no
```
